src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -96,6 +96,11 @@ internal static class RuntimeExceptionHelpers
                         FailFast("Access Violation: Attempted to read or write protected memory. This is often an indication that other memory is corrupt. The application will be terminated since this platform does not support throwing an AccessViolationException.");
                         return null;
 
+#pragma warning disable CS0618 // ExecutionEngineException is obsolete
+                    case ExceptionIDs.IllegalInstruction:
```

Comment:
I am not sure if we want to create an exception on the NativeAOT or behave like in the case of Access violation and just failfast here.

Reply:
I think `PlarformNotSupportedException` would be the best because it will allow us to have same behavior as CoreCLR for valid C#, but invalid use of hardware intrinsics.

For example:

```csharp
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.X86;

Sse41.Blend(Vector128<double>.One, default, default);
```

I'm not guarding this with `Sse41.IsSupported` and the generated code has an unguarded `blendpd` instruction. CoreCLR-JIT would generate a PNSE throw instead of the unguarded `blendpd`.

Reply:
I am not sure whether we want these faults to be catchable exceptions. The JIT assumes that these instructions won't throw, and it may take shortcuts when generating GC info for this IP. Catching and handling this fault would produce a GC hole.

Reply:
In coreclr, it is not catchable - the exception is created just for reporting the failure.

Reply:
> In coreclr, it is not catchable - the exception is created just for reporting the failure.

The exceptions should not be catchable in native aot either. I think it should look more like the access violation handling. I have tried the following with this change on Native AOT (my machine does not have AvxVnni support):
`dotnet publish /p:PublishAot=true /p:IlcInstructionSet=avx2`

```csharp
using System.Runtime.Intrinsics;
using System.Runtime.Intrinsics.X86;

Console.WriteLine(AvxVnni.IsSupported);

try
{
    AvxVnni.MultiplyWideningAndAdd(default(Vector128<int>), default(Vector128<byte>), default(Vector128<sbyte>)).ToString();
}
catch (Exception e)
{
    Console.WriteLine("HANDLED!");
}
```
It prints:
```
False
HANDLED!
```

Reply:
I've updated it to fail fast on NativeAOT like we do for the access violation.